quadra/tasks/classification.py

```diff
@@ -910,17 +1036,36 @@ def test(self) -> None:
 
         image_labels = []
         predicted_classes = []
-
-        with torch.no_grad():
-            for batch_item in tqdm(test_dataloader):
-                im, target = batch_item
-
-                outputs = self.deployment_model(im.to(self.device))
-                probs = torch.softmax(outputs, dim=1)
-                preds = torch.max(probs, dim=1).indices
-
-                predicted_classes.append(preds.tolist())
-                image_labels.extend(target.tolist())
+        grayscale_cams_list = []
+
+        if self.gradcam:
+            self.prepare_gradcam()
+
+        for batch_item in tqdm(test_dataloader):
+            im, target = batch_item
+            im = im.to(self.device).detach()
```

Comment:
I don't like that we compute gradients when is not necessary

with torch.set_grad_enabled(self.gradcam):